File: Decorator/cpp-source/notifiers/LogNotifier.cpp
//
// Created by Pavel Akhtyamov on 2019-04-02.
//

#include <LogNotifier.h>

void LogNotifier::Notify(const std::string &message) {
  stream_ << message << std::endl;
}

LogNotifier::LogNotifier(const std::string &filename) {
  stream_.open(filename, std::ofstream::out | std::ofstream::app);
}

LogNotifier::~LogNotifier() {
  stream_.close();
}

File: Decorator/python-source/notifiers/LogNotifier.py
from __future__ import annotations

from pathlib import Path

from .Notifier import Notifier


class LogNotifier(Notifier):
    """Основа цепочки: пишет сообщение в файл и никого не оборачивает."""

    def __init__(self, filename: str | Path) -> None:
        self._path = Path(filename)

    def notify(self, message: str) -> None:
        with self._path.open("a", encoding="utf-8") as stream:
            stream.write(message + "\n")
